### PR TITLE
[INOYI-33] - added clustering option for the Leaflet map provider

### LINKED ISSUE #2315: Map pins overlapping each other when it placed too close
Sometimes when the map pins are placed too cose to each other we can see this picture:
INOYI-33PinsOverlapping

As solution we can use clustering for map.
For Gmap it standard functionality
For Leaflet we can use this lib https://github.com/Leaflet/Leaflet.markercluster

### PR DESCRIPTION
**Pull request by aleevas**
Original Issue, this PR is going to fix: #2315 

This PR should add the clustering option for the Leaflet map provider.
When you will select the Leaflet provider you can see addition fieldgroup "Group markers" with a few options:
INOYI-33MapClusters

- **Enable**:
Group close markers together on the map
INOYI-33ClustersOnTheMap

- **Hovering over a cluster shows the bounds of its markers.** :
   When you mouse over a cluster it shows the bounds of its markers.
INOYI-33ClustersOnHoverOption

-  **Clicking a cluster zooms to the bounds.**:
When you click a cluster we zoom to its bounds

- **Disable clustering at zoom**:
If set, at this zoom level and below, markers will not be clustered.

**Note** more details about options you can find on [this](https://github.com/Leaflet/Leaflet.markercluster#all-options) page

## Steps for review

- [ ] Log as admin.
- [ ] Go to the OPenY map settings page (admin/openy/settings/openy_map).
- [ ] Please chose  the Leaflet as map provider.
- [ ] You should find the new fieldgroup "Group markers"  under the "Base layer override".
- [ ] By default this option should be disabled (for back compatibility).
- [ ] Enable this option and save form.
- [ ] Go to the locations page and you should see that close locations should be grouped to cluster.
- [ ] Please back to settings form and check all clustering options according to the it description.
- [ ] Also please check searh and filtering options. Its should wotk too.
INOYI-33Filters

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer. See [Code of Conduct](https://github.com/ymcatwincities/openy/wiki/Open-Y-Code-of-Conduct-and-Best-Practices).
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

File: modules/custom/openy_map/config/install/openy_map.settings.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ leaflet:
   base_layer_override:
     enable: 0
     override: ''
+  clustering:
```
Comment (podarok) [inline modules/custom/openy_map/config/install/openy_map.settings.yml:9]: We need to add hook_update_N for supporting upgrade path

**Review by podarok — CHANGES_REQUESTED** [review]

File: modules/custom/openy_map/config/install/openy_map.settings.yml (hunk 1 above)
Comment (aleevas) [reply modules/custom/openy_map/config/install/openy_map.settings.yml:9]: @podarok thank you for quick review.
Like this? Did I get it right? 
```
/**
 * Add Leaflet base layer override configuration.
 */
function openy_map_update_8004() {
  $config_updater = \Drupal::service('openy_upgrade_tool.param_updater');
  $config_dir = drupal_get_path('module', 'openy_map') . '/config/install/';
  $config_name = 'openy_map.settings';
  $config_path = $config_dir . $config_name . '.yml';
  $config_updater->update($config_path, $config_name, 'leaflet.base_layer_override');
}
```

Comment (aleevas) [reply modules/custom/openy_map/config/install/openy_map.settings.yml:9]: @podarok 
I hope i got it right. So I've added hook_update_n in my new commit https://github.com/ymcatwincities/openy/pull/2316/commits/9a80337c1221dbd3faee12396107ca9b627428db

**Comment by podarok** [comment]
ok to test

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2857/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2857/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2857/

**Comment by podarok** [comment]
This works as described 
image

@sarah-halby to decide when we should include features in OpenY due to our policy

**Comment by aleevas** [comment]
@podarok 
sorry, but I found a bug with clustering settings.
I've fixed that in my latest commit https://github.com/ymcatwincities/openy/pull/2316/commits/8f27866c3198112da854ffd66bedc2752b8846b6
Please re-check clustering settings

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2859/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2859/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2859/

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2859/

**Comment by podarok** [comment]
Does work 
image

**Comment by podarok** [comment]
done

**Pull request closed without merging**